### PR TITLE
LPS-89796 Filter by Keyword name.

### PR DESCRIPTION
**Pull request by cgoncas**

### CODE REVIEW

File: modules/apps/headless/headless-foundation/headless-foundation-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -626,6 +626,14 @@ paths:
           name: pageSize
           schema:
             type: integer
+        - in: query
+          name: filter
+          schema:
+            type: string
+        - in: query
+          name: sort
```
Comment (dgarciasarai) [inline modules/apps/headless/headless-foundation/headless-foundation-impl/rest-openapi.yaml]: I think the yaml file should be sort by name

File: modules/apps/headless/headless-foundation/headless-foundation-impl/src/main/java/com/liferay/headless/foundation/internal/odata/entity/v1_0/KeywordEntityModel.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @author Cristina González
+ */
+public class KeywordEntityModel implements EntityModel {
+
+	public KeywordEntityModel() {
+		_entityFieldsMap = Stream.of(
```
Comment (dgarciasarai) [inline modules/apps/headless/headless-foundation/headless-foundation-impl/src/main/java/com/liferay/headless/foundation/internal/odata/entity/v1_0/KeywordEntityModel.java]: dateCreated? dateModified?

**Comment by dgarciasarai** [comment]
Good job @cgoncas! :)

File: modules/apps/headless/headless-foundation/headless-foundation-impl/rest-openapi.yaml (hunk 1 above)
Comment (cgoncas) [reply modules/apps/headless/headless-foundation/headless-foundation-impl/rest-openapi.yaml]: There are more unsorted keys, we will sort it once, with SF.

File: modules/apps/headless/headless-foundation/headless-foundation-impl/src/main/java/com/liferay/headless/foundation/internal/odata/entity/v1_0/KeywordEntityModel.java (hunk 2 above)
Comment (cgoncas) [reply modules/apps/headless/headless-foundation/headless-foundation-impl/src/main/java/com/liferay/headless/foundation/internal/odata/entity/v1_0/KeywordEntityModel.java]: I will do it :), thank you for notice.

**Comment by dgarciasarai** [comment]
ci:test:sf

File: modules/apps/headless/headless-foundation/headless-foundation-impl/src/main/java/com/liferay/headless/foundation/internal/resource/v1_0/KeywordResourceImpl.java [hunk 3]
```diff
@@ -53,15 +71,46 @@ public boolean deleteKeyword(Long keywordId) throws Exception {
...
+			filter, _indexerRegistry.nullSafeGetIndexer(AssetTag.class),
+			pagination,
+			booleanQuery -> {
+			},
+			queryConfig -> {
+				queryConfig.setSelectedFieldNames(Field.ASSET_TAG_IDS);
+			},
+			searchContext -> {
+				searchContext.setAttribute(
+					Field.CLASS_NAME_ID, className.getClassNameId());
```
Comment (victorg1991) [inline modules/apps/headless/headless-foundation/headless-foundation-impl/src/main/java/com/liferay/headless/foundation/internal/resource/v1_0/KeywordResourceImpl.java]: I think this is not need, I was going to review other resource to remove it

**Comment by victorg1991** [comment]
👏 Perfect!

**Comment by victorg1991** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by victorg1991** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Sent to @brianchandotcom: https://github.com/brianchandotcom/liferay-portal/pull/68239

**Pull request closed without merging**